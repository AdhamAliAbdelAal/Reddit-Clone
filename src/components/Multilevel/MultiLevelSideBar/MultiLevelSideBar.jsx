import { useCookies } from 'react-cookie';
import { useState, useEffect } from 'react';

// Components
import SideBar from '../../SideBar/SideBar';
import SideBarSubreddit from '../../SubReddit/SideBar/SideBar';
import UserInfo from '../../MainProfile/Profile/MainContent/Sidebar/UserInfo/UserInfo';
import OtherProfileUserInfo from '../../MainProfile/OtherProfile/OtherProfileMainContent/OtherProfileSidebar/OtherProfileUserInfo/OtherProfileUserInfo';
import BackToTop from '../../BackToTop/BackToTop';

// Contexts
import { usePostContext } from '../../../contexts/PostContext';
import UserProvider from '../../../contexts/UserProvider';

// Server
import getSubredditAllData from '../../SubReddit/SubrridetDataServer';

function MultiLevelSideBar() {
  // Context
  const { post } = usePostContext();

  // cookies
  const [cookies] = useCookies(['redditUser']);

  // useState
  const [sunreddit, setSubreddit] = useState(null);

  // Fetch Subreddit
  // eslint-disable-next-line no-unused-vars
  const [SubredditData, dataError, statusCode] = getSubredditAllData(post?.owner?.name);

  useEffect(() => {
    setSubreddit(SubredditData);
  }, [SubredditData]);

  return (
    post && (
      post?.ownerType === 'Subreddit' ? (
        <SideBarSubreddit
          disc={sunreddit?.description}
          topics={sunreddit?.topics}
          Name={sunreddit?.name}
          primaryTopic={sunreddit?.primaryTopic}
          createdAt={sunreddit?.createdAt}
          moderatoesName={sunreddit?.moderators}
          username={cookies?.redditUser?.username}
          members={sunreddit?.membersCount}
          postCard
        />
      )
        : (
          <SideBar responsive>
            {(post?.ownerType === 'User'
              ? ((cookies?.redditUser?.id === post?.owner?._id)
                ? (
                  <UserProvider name={post?.owner?.name}>
                    <UserInfo />
                  </UserProvider>
                )
                : (
                  <UserProvider name={post?.owner?.name}>
                    <OtherProfileUserInfo />
                  </UserProvider>
                )
              ) : null)}
            <div style={{ height: '50px' }} />
            <BackToTop />
          </SideBar>
        )
    ));
}

export default MultiLevelSideBar;
