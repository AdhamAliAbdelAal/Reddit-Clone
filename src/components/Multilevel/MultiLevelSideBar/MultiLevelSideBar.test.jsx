import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';

import PostContextProvider from '../../../contexts/PostContext';
import MultiLevelSideBar from './MultiLevelSideBar';

// test snapshot
test('MultiLevelSideBar-test-snapshot', async () => {
  const tree = renderer.create(
    <PostContextProvider postID="123556">
      <MultiLevelSideBar />
    </PostContextProvider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
