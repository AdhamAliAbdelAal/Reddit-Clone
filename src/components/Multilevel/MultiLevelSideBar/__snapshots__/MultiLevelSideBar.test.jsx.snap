// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiLevelSideBar-test-snapshot 1`] = `null`;
